### PR TITLE
[Chore] Before fn for tests + Destructure Bundled args in contract

### PR DESCRIPTION
**Pull request by narlypunk**
- Added `before` fn for tests that adds mock ERC20 tokens to accounts and sample NFTs
- Passed `Bundle` as arg to `createOffer` and destructured in function due to `Stack Too Deep` error on contract compilation

### CODE REVIEW

**Comment by narlypunk** [comment]
@doylio @devlyn37 lmk if this looks good to merge

File: test/index.ts [hunk 1]
```diff
@@ -1,19 +1,52 @@
 import { expect } from "chai";
 import { ethers } from "hardhat";
 
-describe("Greeter", function () {
-  it("Should return the new greeting once it's changed", async function () {
-    const Greeter = await ethers.getContractFactory("Greeter");
-    const greeter = await Greeter.deploy("Hello, world!");
-    await greeter.deployed();
+describe("BarterMarket", function () {
+  before(async function () {
```
Comment (devlyn37) [inline test/index.ts]: I think this should be before each for a fresh slate each test

File: test/index.ts [hunk 2]
```diff
@@ -1,19 +1,52 @@
...
+    // TokenIDs: 1 - 5
+    for (var i = 0; i < 5; i++) {
+      await ERC721.connect(account1).mint();
+    }
+
+    // TokenIDs: 6 - 15
+    for (var i = 0; i < 10; i++) {
+      await ERC721.connect(account2).mint();
+    }
+
```
Comment (devlyn37) [inline test/index.ts:37]: I'm not sure if this is happens when the signers are created, but it would probably be useful to give each account some eth

**Review by devlyn37 — COMMENTED** [review]
Nice! left a couple comments

File: test/index.ts (hunk 1 above)
Comment (narlypunk) [reply test/index.ts]: Changed this to `beforeEach`

File: test/index.ts (hunk 2 above)
Comment (narlypunk) [reply test/index.ts:37]: Hardhat adds a default balance to these accounts :)

**Pull request merged**